### PR TITLE
BE: design availability

### PR DESCRIPTION
**Pull request by davidleecodes**
### What this PR does (required):
- Added availability to profile model

### Screenshots / Videos (required):
Screenshot 2021-06-21 152550
Screenshot 2021-06-21 152615

### Any information needed to test this feature (required):
- Post what steps to take to be able to test this feature (eg. "sign up as a new user and upload a profile picture from the navbar")

### Any issues with the current functionality (optional):
- List any problems where you were not able to complete all tasks on ticket
- List any problems this PR may cause for the project or other active PRs
- Post a screenshot/video of the problem, along with detailed console outputs where applicable

### CODE REVIEW

**Review by rajivtitus — APPROVED** [review]
Lgtm! 

Please resolve conflicts before merging.

**Review by rajivtitus — APPROVED** [review]

File: server/routes/profile.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+const protect = require("../middleware/auth");
+const {
+  newUser,
+  editProfile,
+  getProfile,
+  findSitters,
+} = require("../controllers/profile");
+
+router.route("/new").post(protect, newUser);
+router.route("/editprofile/:id").patch(protect, editProfile);
```
Comment (bonnieli) [inline server/routes/profile.js:12]:
```suggestion
router.route("/editProfile/:id").patch(protect, editProfile);
```

best to keep things consistent even for URLs for readability sake as well

File: server/routes/profile.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+router.route("/getprofile/:id").get(protect, getProfile);
```
Comment (bonnieli) [inline server/routes/profile.js:13]:
```suggestion
router.route("/getProfile/:id").get(protect, getProfile);

```

File: server/controllers/profile.js [hunk 3]
```diff
@@ -1,97 +1,117 @@
...
-        const updateProfile = await Profile.findOneAndUpdate({userId: userId});
-            if(updateProfile) { 
-                profile.save(res.status(200).json({profile: updateProfile})
-            )} else {
-                res.status(404).json({ message: "Profile not found"})
-            }
-        } catch (error) {
-            return res.status(500).json({error: "Could not update profile"})
-        };
+  const userId = req.params.id;
```
Comment (bonnieli) [inline server/controllers/profile.js:64]: for this controller, will we ever need `/editProfile` for other users? meaning will other users be able to edit other user's profiles?

from my assumptions, likely not. in which we should not open this endpoint to the public. We'll remove `/:id` from the controller and access the logged in user ID through cookies instead :)

**Pull request merged**